lib/model/folder_sendrecv.go

```diff
@@ -204,10 +204,10 @@ func (f *sendReceiveFolder) pull() bool {
 			// we're not making it. Probably there are write
 			// errors preventing us. Flag this with a warning and
 			// wait a bit longer before retrying.
-			if folderErrors := f.PullErrors(); len(folderErrors) > 0 {
+			if folderPullErrors := f.PullErrors(); len(folderPullErrors) > 0 {
 				events.Default.Log(events.FolderErrors, map[string]interface{}{
 					"folder": f.folderID,
-					"errors": folderErrors,
+					"errors": folderPullErrors,
```

Comment:
This is not nice, folders errors which is generic returns pull errors only.
Perhaps we should enrich FileError to define error type, rename APIs to talk about folder/errors rather than folder/pullErrors and have a single consolidated UI with a single consolidated set of errors for both pull and download, which gets rid of all the switch if hacks in the UI and simplifies it there.